orienteer-core/src/main/java/org/orienteer/core/component/visualizer/TagsVisualizer.java

```diff
@@ -0,0 +1,108 @@
+import ru.ydn.wicket.wicketorientdb.model.DynamicPropertyValueModel;
+import ru.ydn.wicket.wicketorientdb.model.OQueryModel;
+import ru.ydn.wicket.wicketorientdb.utils.OChoiceRenderer;
+
+import com.orientechnologies.orient.core.db.record.OIdentifiable;
+import com.orientechnologies.orient.core.metadata.schema.OClass;
+import com.orientechnologies.orient.core.metadata.schema.OProperty;
+import com.orientechnologies.orient.core.metadata.schema.OType;
+import com.orientechnologies.orient.core.record.impl.ODocument;
+
+/**
+ * {@link IVisualizer} to show links as tags control
+ */
+public class TagsVisualizer extends AbstractSimpleVisualizer {
+    public TagsVisualizer() {
+        super("tags", false, OType.LINK, OType.LINKLIST, OType.LINKSET, OType.LINKBAG);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <V> Component createComponent(String id, DisplayMode mode,
+                                         IModel<ODocument> documentModel, IModel<OProperty> propertyModel, IModel<V> valueModel) {
+        if (DisplayMode.EDIT.equals(mode)) {
+            OProperty property = propertyModel.getObject();
+            if (property.getType().isMultiValue()) {
+                return new Select2MultiChoice<ODocument>(id, (IModel<Collection<ODocument>>) valueModel, new ODocumentChoiceProvider(propertyModel));
+            } else {
+                return new Select2Choice<ODocument>(id, (IModel<ODocument>) valueModel, new ODocumentChoiceProvider(propertyModel));
+            }
+        } else {
+            return null;
+        }
+    }
+
+    public class ODocumentChoiceProvider extends ChoiceProvider<ODocument> {
+
+        private transient IOClassIntrospector oClassIntrospector;
+
+        public IModel<OProperty> propertyModel;
+
+        public ODocumentChoiceProvider(IModel<OProperty> propertyModel) {
+            this.propertyModel = propertyModel;
+        }
+
+        protected IOClassIntrospector getOClassIntrospector() {
+            if (oClassIntrospector == null) {
+                oClassIntrospector = OrienteerWebApplication.get().getOClassIntrospector();
+            }
+            return oClassIntrospector;
+        }
+
+        @Override
+        public String getDisplayValue(ODocument document) {
+            return getOClassIntrospector().getDocumentName(document);
+        }
+
+        @Override
+        public String getIdValue(ODocument document) {
+            return document.getIdentity().toString();
+        }
+
+        @Override
+        public void query(String query, int i, Response<ODocument> response) {
+            OProperty property = propertyModel.getObject();
+            OClass oClass = property.getLinkedClass();
+            StringBuilder sql = new StringBuilder("SELECT FROM " + oClass.getName());
+            if (!Strings.isEmpty(query)) {
+                sql.append(" WHERE " + getOClassIntrospector().getNameProperty(oClass).getName() + " LIKE '%" + query + "%'");
```

Comment:
Query is the same from call to call. Recommend to prepare it in constructor.
Plus there is special function CONTAINSTEXT for checking like we needed in.